lib/nostrum/cache/guild/guild_server.ex

```diff
@@ -442,4 +447,14 @@ defmodule Nostrum.Cache.Guild.GuildServer do
     {:reply, {guild_id, old_emojis, emojis}, %{state | emojis: emojis}}
   end
 
+  def handle_call({:update, :voice_state, guild_id, voice_state}, _from, state) do
+    old_voice_states = state.voice_states
+    new_voice_states =
+      [
+        voice_state |
+        state.voice_states
+        |> Enum.reject(fn(map) -> map.user_id == voice_state.user_id end)
+      ]
+    {:reply, {guild_id, old_voice_states, new_voice_states}, %{state | voice_states: new_voice_states}}
```

Comment:
With the list, one can tell other's states in the guild, but with only the single state not so much information will be provided (It can, if the user keeps track on all the states, but that's what nostrum does for him, so it would be ridiculous), so I think the former will be good enough for now...